### DISCUSSION TITLE
Using cross-compilation with autodetecting target does not work

### DISCUSSION THREAD

**Discussion by PPQQXX** [openssl/openssl#21387] [Q&A] [answered]
![image](https://github.com/openssl/openssl/assets/32433430/719eb29e-0617-411d-a919-e8954ba35be5)
my environment: OpenSSL 1.0.2g  1 Mar 2016
what should i do?

but when i use openssl-3.0.8,if i input  ./config，the tips like this 
![image](https://github.com/openssl/openssl/assets/32433430/86a321a1-d6dd-4833-8382-41dabf05ac27)

**Comment by PPQQXX** [comment]
I find "Configuring OpenSSL version 1.1.1t (0x1010114fL) for linux-x86_64-" and "Configuring OpenSSL version 1.1.1t (0x1010114fL) for linux-x86_64",it seems the "linux-x86_64-" not match tips?

**Comment by t8m** [comment, marked as answer]
You cannot use the config target autodetection when doing cross-compilation, because the autodetection uses the build host system information to guess the config target. You need to explicitly specify the config target. I assume it would be `linux-armv4` or `linux-aarch64` in your case.

**Reply by PPQQXX** [reply]
thanks for your help.
